developer-hub/src/polkadot_sdk/cumulus.rs

```diff
@@ -0,0 +1,132 @@
+//! # Cumulus
+//!
+//! Substrate provides a framework through which a blockchain node and runtime can easily be
+//! created. Cumulus aims to extend the same approach to creation of Polkadot parachains.
+//!
+//! > Cumulus clouds are shaped sort of like dots; together they form a system that is intricate,
+//! > beautiful and functional.
+//!
+//! ## Example: Runtime
+//!
+//! A cumulus based runtime is fairly similar to a normal [FRAME]-based runtime. Most notably, the
+//! following changes are applied to a normal FRAME-based runtime to make it a cumulus-based
+//! runtime:
+//!
+//! #### Cumulus Pallets
+//!
+//! A parachain runtime should use a number of pallets that are provided by Cumulus. Notably:
+//!
+//! - [`frame-system`], like all FRAME-based runtimes.
+//! - [`cumulus-pallet-parachain-system`]
+//! - [`parachain-info`]
+#![doc = docify::embed!("./src/polkadot_sdk/cumulus.rs", system_pallets)]
+//!
+//! Given that all cumulus-based runtimes use a simple aura-based consensus mechanism, the following
+//! pallets also need to be added:
+//!
+//! - [`pallet-timestamp`]
+//! - [`pallet-aura`]
+//! - [`cumulus-pallet-aura-ext`]
+// #![doc = docify::embed!("./src/lib.rs", consensus_pallets)]
+//!
+//!
+//! Finally, a separate macro, similar to `impl_runtime_api`, which create the default set of
+//! runtime apis, will generate the parachain runtime's main additional runtime api, also known as
+//! PVF.
+#![doc = docify::embed!("./src/polkadot_sdk/cumulus.rs", validate_block)]
+//!
+//!
+//! ## Example: Running a node
+//!
+//! TODO
```

Comment:
Awesome. It could be worth shouting out Zombienet?

Reply:
Yeah maybe, or whatever is the simplest way to have an example that showcases the next step in a "guaranteed to work way". Maybe something like a CLI test? 

I have the same dilemma in the currency tutorial as well. The next demo step I want craft is adding this pallet to a runtime, adding that runtime to a node, running it. Not sure how to demonstrate that in our framework yet, but determined to make it work.

Reply:
For the last year or so I have been using our `small_network` specification to spawn up simple tests quickly.
```
zombienet --provider native spawn ./cumulus/zombienet/examples/small_network.toml
```

Downside if of course that it does things "behind the scene" that would be useful for the average user to know. Like getting to know the most basic CLI args.

Upside is that it introduces new users directly to some tooling that will be useful later on.

Reply:
I don't think Zombienet is the right tool to explain this. I would see it as a good plan B if nothing else works. The main issue I see with it is that now you have to learn two new things. 

As a plan A, I highly recommend exploring CLI tests as a way to "docify bash commands". If this fails, we can reside to zombienet or something else.